### PR TITLE
Healthcheck - round start

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala [hunk 1]
```diff
@@ -91,37 +93,86 @@ abstract class HealthCheckConsensus[
...
 
   def startRound(
     key: HealthCheckKey,
     peers: Set[Peer],
-    roundId: RoundId,
+    roundId: HealthCheckRoundId,
     roundType: HealthCheckRoundType
-  ): F[Unit] = ???
+  ): F[Unit] =
+    Clock[F].monotonic
```
Comment (tbekas) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:112]: 1. If you need an actual time of the start of the round then `Clock[F].realTime` should be used instead of `monotonic`
2. It looks like you're not using the `startedAt` paramter at all.

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala [hunk 2]
```diff
@@ -91,37 +93,86 @@ abstract class HealthCheckConsensus[
...
+      .map(_.toSeconds)
+      .flatMap { startedAt =>
+        allRounds.get.map {
+          case ConsensusRounds(historical, inProgress) =>
+            def inProgressRound = inProgress.get(key)
+            def historicalRound = historical.find(_.roundId == roundId)
+
+            historicalRound.orElse(inProgressRound).map(_ => none).getOrElse {
+              HealthCheckRound.make[F, A, B]().some
```
Comment (tbekas) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:121]: This will result in `Option[Option[A]]`. Is the nested option necessary? It makes reasoning about the code difficult to follow.

**Review by tbekas — CHANGES_REQUESTED** [review]

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala (hunk 1 above)
Comment (marcinwadon) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:112]: I'm going to pass it to the round and most likely I need monotonic instead of realTime. If so, I'll change that in next PR with actual usage in a particular round

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala (hunk 2 above)
Comment (marcinwadon) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:121]: Yes, I think so.
tl;dr;
It checks if historical or inProgress round exists, and if no - creates a new round declaration.
It returns the Option (to indicate whether new round should be created or no) and the value is `F[Round]`
so it's not `Option[Option[A]]` but `Option[F[A]]`

Comment (tbekas) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:121]: Yep, you're right, the final type of this expression is `F[Option[F[A]]]`.

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala (hunk 1 above)
Comment (tbekas) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala:112]: Ok, it's difficult for reviewers to tell the difference between what's not used yet and will be used later on and what's not used because the author forgot to remove it.

**Review by tbekas — APPROVED** [review]
